java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/project/RemotePlatformImpl.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.openjdk.project;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.io.File;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.swing.event.ChangeListener;
+import org.netbeans.api.annotations.common.NonNull;
+import org.netbeans.modules.java.openjdk.project.JDKProject.Root;
+import org.netbeans.modules.java.openjdk.project.JDKProject.RootKind;
+import org.netbeans.spi.java.source.RemotePlatform;
+import org.netbeans.spi.project.ProjectConfigurationProvider;
+import org.openide.filesystems.FileChangeAdapter;
+import org.openide.filesystems.FileChangeListener;
+import org.openide.filesystems.FileEvent;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.util.ChangeSupport;
+import org.openide.util.Lookup;
+import org.openide.util.lookup.Lookups;
+
+/**
+ *
+ * @author lahvac
+ */
+public class RemotePlatformImpl implements RemotePlatform {
+
+    private static final Map<FileObject, RemotePlatformImpl> jdkRoot2Platform = new HashMap<>();
+
+    public static @NonNull RemotePlatformImpl getProvider(FileObject jdkRoot, ConfigurationImpl.ProviderImpl configurations) {
+        return jdkRoot2Platform.computeIfAbsent(jdkRoot, r -> new RemotePlatformImpl(configurations));
+    }
+
+    private final ChangeSupport cs = new ChangeSupport(this);
+    private final ConfigurationImpl.ProviderImpl configurations;
+    private File modules;
+    private FileChangeListener modulesListener = new FileChangeAdapter() {
+        @Override
+        public void fileChanged(FileEvent fe) {
+            cs.fireChange();
+        }
+    };
+
+    public RemotePlatformImpl(ConfigurationImpl.ProviderImpl configurations) {
+        this.configurations = configurations;
+        this.configurations.addPropertyChangeListener(new PropertyChangeListener() {
```

Comment:
Jackpot:
warning: This anonymous inner class creation can be turned into a lambda expression.